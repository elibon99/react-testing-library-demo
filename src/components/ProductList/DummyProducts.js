import Macbook from "../../assets/images/Macbook.jpg"
import Iphone from "../../assets/images/Iphone.jpg"
import AirPods from "../../assets/images/AirPods.webp"

export const DummyProducts = [
    {
        id: 1,
        title: "Macbook",
        price: 16000,
        image: Macbook
    },
    {
        id: 2,
        title: "Iphone",
        price: 7000,
        image: Iphone
    },
    {
        id: 3,
        title: "AirPods",
        price: 1500,
        image: AirPods
    }
]