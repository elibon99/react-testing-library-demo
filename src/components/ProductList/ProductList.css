
.product-list-container {
    display: flex;
    flex-direction: column;
    background: white;
    border-radius: 8px;
    width: 65%;
    padding: 30px 30px;
}

.product-list {
    display: flex;
    justify-content: space-between;
    margin-top: 20px;
    align-items: stretch;
}

.product-list-header {
    margin: 0 0;
}