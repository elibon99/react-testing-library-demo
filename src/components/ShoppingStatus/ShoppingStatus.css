.shopping-status-container {
    display: flex;
    flex-direction: column;
    margin-top: auto;
}

.shopping-status-title {
    margin: 0 0;
    color: #424242;
    font-size: 16px;
}

.shopping-status-title:nth-child(even) {
    margin-top: 10px;
}