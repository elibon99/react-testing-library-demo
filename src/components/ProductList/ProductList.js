import React from 'react'
import "./ProductList.css"
import {DummyProducts} from "./DummyProducts";
import ProductItem from "../ProductItem/ProductItem";

const ProductList = ({products, setProducts}) => {
    const addProductToCart = (id) => {
        let product = DummyProducts.find((product) => product.id === id)
        setProducts(prev => [...prev, product])
    }

    return (
        <div className="product-list-container">
            <h3 className="product-list-header">Products</h3>
            <div className="product-list" data-testid={"product-list"}>
                {DummyProducts.map((product, index) => {
                    return(
                        <ProductItem key={index} product={product} products={products} addProductToCart={addProductToCart}/>
                    )
                })}
            </div>
        </div>
    )
}

export default ProductList