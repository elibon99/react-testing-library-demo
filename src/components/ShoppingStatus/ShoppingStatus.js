import React from 'react'
import "./ShoppingStatus.css"

const ShoppingStatus = ({products}) => {
    const calculatePrice = () => {
        let price = 0
        for (let product of products) {
            price += product.price
        }
        return price
    }

    return (
        <div className="shopping-status-container">
            <h3 className="shopping-status-title">Total items: {products.length}</h3>
            <h3 className="shopping-status-title">Total price: {calculatePrice()}</h3>
        </div>
    )
}

export default ShoppingStatus
