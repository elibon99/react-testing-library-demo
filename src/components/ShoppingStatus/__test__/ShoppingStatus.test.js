import {render, screen} from "@testing-library/react";
import ShoppingStatus from "../ShoppingStatus";
import {DummyProducts} from "../../ProductList/DummyProducts";

const ShoppingCart = DummyProducts.filter((product) => product.title === "Macbook")

describe('<ShoppingStatus/>', () => {

    /**
     * Checks that ShoppingStatus is correctly rendered
     */
    it('renders correctly', async () => {
        const {asFragment} = render(
            <ShoppingStatus
                products={ShoppingCart}
            />
        )
        expect(asFragment()).toMatchSnapshot()
    })

    /**
     * Checks that ShoppingStatus renders the correct amount of items and
     * total price
     */
    it('should render the correct amount of items and total price', async () => {
        render(
            <ShoppingStatus
                products={ShoppingCart}
            />);
        const totalItemsElement = screen.getByText(/Total items: 1/i);
        const totalPriceElement = screen.getByText(/Total price: 16000/i);
        expect(totalItemsElement).toBeInTheDocument();
        expect(totalPriceElement).toBeInTheDocument();
    })
})

