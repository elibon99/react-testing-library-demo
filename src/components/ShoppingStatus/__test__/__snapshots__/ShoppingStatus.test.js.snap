// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShoppingStatus/> renders correctly 1`] = `
<DocumentFragment>
  <div
    class="shopping-status-container"
  >
    <h3
      class="shopping-status-title"
    >
      Total items: 1
    </h3>
    <h3
      class="shopping-status-title"
    >
      Total price: 16000
    </h3>
  </div>
</DocumentFragment>
`;
